React/Base/RCTBridge.m

```diff
@@ -161,21 +161,18 @@ - (void)bindKeys
 {
   RCTAssertMainQueue();
 
-  [[NSNotificationCenter defaultCenter] addObserver:self
-                                           selector:@selector(reload)
-                                               name:RCTReloadNotification
-                                             object:nil];
-
 #if TARGET_IPHONE_SIMULATOR
   RCTKeyCommands *commands = [RCTKeyCommands sharedInstance];
 
   // reload in current mode
+  __weak typeof(self) weakSelf = self;
   [commands registerKeyCommandWithInput:@"r"
                           modifierFlags:UIKeyModifierCommand
                                  action:^(__unused UIKeyCommand *command) {
-    [[NSNotificationCenter defaultCenter] postNotificationName:RCTReloadNotification
-                                                        object:nil
-                                                      userInfo:nil];
+    __strong typeof(self) strongSelf = weakSelf;
```

Comment:
No need to do strongify here. Just do `[weakSelf requestReload]`. If weakSelf is nil, this won't crash.